gtk4/src/constraint_guide.rs

```diff
@@ -6,43 +6,70 @@ use glib::IsA;
 
 pub trait ConstraintGuideExtManual {
     #[doc(alias = "gtk_constraint_guide_get_max_size")]
-    fn get_max_size(&self, width: i32, height: i32);
+    fn get_max_size(&self) -> (Option<i32>, Option<i32>);
 
     #[doc(alias = "gtk_constraint_guide_get_min_size")]
-    fn get_min_size(&self, width: i32, height: i32);
+    fn get_min_size(&self) -> (Option<i32>, Option<i32>);
 
     #[doc(alias = "gtk_constraint_guide_get_nat_size")]
-    fn get_nat_size(&self, width: i32, height: i32);
+    fn get_nat_size(&self) -> (Option<i32>, Option<i32>);
 }
 
 impl<O: IsA<ConstraintGuide>> ConstraintGuideExtManual for O {
-    fn get_max_size(&self, width: i32, height: i32) {
+    fn get_max_size(&self) -> (Option<i32>, Option<i32>) {
         unsafe {
-            ffi::gtk_constraint_guide_get_max_size(
-                self.as_ref().to_glib_none().0,
-                width as *mut _,
-                height as *mut _,
-            );
+            let width = std::ptr::null_mut::<i32>();
+            let height = std::ptr::null_mut::<i32>();
+            ffi::gtk_constraint_guide_get_max_size(self.as_ref().to_glib_none().0, width, height);
```

Comment:
This is wrong. You must pass a pointer to an integer here.

```rust
let mut width = MaybeUninit::uninit();
let mut height = MaybeUninit::uninit();
ffi::gtk_constraint_guide_get_max_size(self.as_ref().to_glib_none().0, width.as_mut_ptr(), height.as_mut_ptr());
let width = width.assume_init();
let height = height.assume_init();
```

The whole thing also can't return `None`. It will always return some kind of integer for both

Reply:
Ah, the annotations mislead me